lib/src/main/java/com/otaliastudios/transcoder/internal/codec/Decoder.kt

```diff
@@ -283,7 +286,13 @@ class Decoder(
 
     override fun release() {
         log.i("release(): releasing codec. dequeuedInputs=$dequeuedInputs dequeuedOutputs=$dequeuedOutputs")
-        codec.stop()
-        codec.release()
+        try {
+            codec.stop()
+            codec.release()
+        }
+        catch (e : Exception) {
+            eventListener?.onDecoderReleaseFailure(codec.name, format, e)
+            throw e
```

Comment:
why throw the exception here?

Reply:
So that it reaches thumbnail engine and it can use MMR as fallback